torchvision/prototype/features/_bounding_box.py

```diff
@@ -69,3 +70,140 @@ def to_format(self, format: Union[str, BoundingBoxFormat]) -> BoundingBox:
+
+        if padding_mode not in ["constant"]:
+            raise ValueError(f"Padding mode '{padding_mode}' is not supported with bounding boxes")
+
+        output = _F.pad_bounding_box(self, padding, format=self.format)
+
+        # Update output image size:
+        # TODO: remove the import below and make _parse_pad_padding available
+        from torchvision.transforms.functional_tensor import _parse_pad_padding
+
+        left, top, right, bottom = _parse_pad_padding(padding)
+        height, width = self.image_size
+        height += top + bottom
+        width += left + right
+
+        return BoundingBox.new_like(self, output, image_size=(height, width))
+
+    def rotate(
+        self,
+        angle: float,
+        interpolation: InterpolationMode = InterpolationMode.NEAREST,
+        expand: bool = False,
+        fill: Optional[List[float]] = None,
+        center: Optional[List[float]] = None,
+    ) -> BoundingBox:
+        from torchvision.prototype.transforms import functional as _F
+
+        output = _F.rotate_bounding_box(
+            self, format=self.format, image_size=self.image_size, angle=angle, expand=expand, center=center
+        )
+        # TODO: update output image size if expand is True
+        if expand:
+            raise RuntimeError("Not yet implemented")
+        return BoundingBox.new_like(self, output, dtype=output.dtype)
+
+    def affine(
+        self,
+        angle: float,
+        translate: List[float],
+        scale: float,
+        shear: List[float],
+        interpolation: InterpolationMode = InterpolationMode.NEAREST,
+        fill: Optional[List[float]] = None,
+        center: Optional[List[float]] = None,
+    ) -> BoundingBox:
+        from torchvision.prototype.transforms import functional as _F
+
+        output = _F.affine_bounding_box(
+            self,
+            self.format,
+            self.image_size,
+            angle,
+            translate=translate,
+            scale=scale,
+            shear=shear,
+            center=center,
+        )
+        return BoundingBox.new_like(self, output, dtype=output.dtype)
+
+    def perspective(
+        self,
+        perspective_coeffs: List[float],
+        interpolation: InterpolationMode = InterpolationMode.BILINEAR,
+        fill: Optional[List[float]] = None,
+    ) -> BoundingBox:
+        from torchvision.prototype.transforms import functional as _F
+
+        output = _F.perspective_bounding_box(self, self.format, perspective_coeffs)
+        return BoundingBox.new_like(self, output, dtype=output.dtype)
+
+    def erase(self, i: int, j: int, h: int, w: int, v: torch.Tensor) -> BoundingBox:
+        raise TypeError("Erase transformation does not support bounding boxes")
+
+    def mixup(self, lam: float) -> BoundingBox:
+        raise TypeError("Mixup transformation does not support bounding boxes")
+
+    def cutmix(self, box: Tuple[int, int, int, int], lam_adjusted: float) -> BoundingBox:
```

Comment:
I don't believe these operations should be kernels. Mixup, Cutmix etc are augmentation strategies. I would even be inclined not to add erase and let people to access the functional directly. The rational is that we should keep the kernels low. As we keep adding augmentations, we do this on the Transforms side. Thoughts?